### DISCUSSION TITLE
How Does Go Serve The Images?

### DISCUSSION THREAD

**Discussion by alteredorange** [imgproxy/imgproxy#845] [Q&A] [answered]
As always, thanks for the awesome work on imgproxy! I'm trying to implement a very basic image proxy in NodeJS and have a question. Is imgproxy streaming the image from the server? or is it storing it on disk "temporarily", and then the image gets deleted right after it's served? (hence the need for a CDN). I can't see anything about saving to file or disk space, so maybe it's the former. Any details/help would be greatly appreciated! Thanks again!

**Comment by DarthSim** [comment, marked as answer]
Thanks for your kind words!
imgproxy downloads originals to the memory.

**Reply by alteredorange** [reply]
Thanks, I think I got a bare bones in nodeJS, these buffers are a pain :) Keep up the good work, imgproxy is the bomb!
